Eatery Blue/AppDelegate.swift

```diff
@@ -12,17 +12,30 @@ import Kingfisher
 import SnapKit
 import Tactile
 import UIKit
+import FirebaseMessaging
 
-class AppDelegate: UIResponder, UIApplicationDelegate {
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
 
     static let shared = UIApplication.shared.delegate as! AppDelegate
 
     private(set) lazy var coreDataStack = CoreDataStack()
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+
         FirebaseApp.configure()
 
+        // Request notification permissions
+        UNUserNotificationCenter.current().delegate = self
+        let authOptions: UNAuthorizationOptions = [.alert, .badge, .sound]
+        UNUserNotificationCenter.current().requestAuthorization(options: authOptions) { granted, error in
+            print("Notification permissions granted: \(granted)")
```

Comment:
One small thing is that rather than using a print statement, I would lean towards using a logger and you can directly just create one in the AppDelegate using OSLog:

```
extension Logger {
    private static var subsystem = Bundle.main.bundleIdentifier!

    static let notifications = Logger(subsystem: subsystem, category: "notifications")
}
```

Reply:
If its an error use
`Logger.notifications.error("TEXT HERE")`

Otherwise use
`Logger.notifications.info("TEXT HERE")`